### PR TITLE
feat(grpc): support connect grpc server via ssl tunnel

### PR DESCRIPTION
**Pull request by mapleeit**

### CODE REVIEW

File: jina/peapods/networking.py [hunk 1]
```diff
@@ -451,6 +451,50 @@ async def task_wrapper(requests, stubs):
...
+        :param options: A list of options to pass to the grpc channel
+
+        :return: A async grpc channel
+        """
+        if options == None:
+            options = GrpcConnectionPool.get_default_grpc_options()
+
+        port = address.split(':')[-1]
+
+        if port == '443':
```
Comment (JoanFM) [inline jina/peapods/networking.py:491]: what is this?

Comment (mapleeit) [reply jina/peapods/networking.py:491]: Using SSL channel if the port is 443, which is dedicated port for HTTPS.

Comment (JoanFM) [reply jina/peapods/networking.py:491]: it should be an option and not a hack like this I would say

Comment (mapleeit) [reply jina/peapods/networking.py]: addressed

File: jina/peapods/networking.py [hunk 2]
```diff
@@ -451,6 +451,39 @@ async def task_wrapper(requests, stubs):
...
+        """
+
+        secure_channel = grpc.secure_channel
+        insecure_channel = grpc.insecure_channel
+
+        if asyncio:
+            secure_channel = grpc.aio.secure_channel
+            insecure_channel = grpc.aio.insecure_channel
+
+        if options == None:
```
Comment (JoanFM) [inline jina/peapods/networking.py:479]: options is

File: jina/peapods/networking.py [hunk 3]
```diff
@@ -562,18 +601,25 @@ def get_default_grpc_options():
         ]
 
     @staticmethod
-    async def send_request_async(request: Request, target: str, timeout=1.0) -> Request:
+    async def send_request_async(
+        request: Request,
+        target: str,
+        timeout=1.0,
```
Comment (JoanFM) [inline jina/peapods/networking.py:607]: please add proper typehints.

**Review by JoanFM — COMMENTED** [review]
For the args u will need to add it to the corresponding parser. We may be better to take from ot

**Pull request closed without merging**